pytest_base_url/plugin.py

```diff
@@ -21,13 +24,21 @@ def base_url(request):
 def _verify_url(request, base_url):
     """Verifies the base URL"""
     verify = request.config.option.verify_base_url
+    verify_timeout = request.config.getoption('verify_base_url_attempts')
     if base_url and verify:
-        response = requests.get(base_url, timeout=10)
-        if not response.status_code == requests.codes.ok:
-            raise pytest.UsageError(
-                'Base URL failed verification!'
-                '\nURL: {0}, Response status code: {1.status_code}'
-                '\nResponse headers: {1.headers}'.format(base_url, response))
+        session = requests.Session()
+        retries = Retry(total=verify_timeout,
+                        backoff_factor=0.1,
+                        status_forcelist=[500, 502, 503, 504])
+        session.mount(base_url, HTTPAdapter(max_retries=retries))
+        try:
+            session.get(base_url)
+        except Exception:
```

Comment:
As I suspected, this is too broad. For example, it's catching when the URL is invalid (missing schema), and giving a misleading message. Looking at the potential exceptions I see `requests.exceptions.RetryError` when retries is exceeded, and the message includes the details such as 'too many 500 error responses' and stderr includes the individual responses. When the connection can't be established I see `requests.exceptions.ConnectionError` with a message 'nodename nor servname provided, or not known'. I think these are clear enough, personally, so we don't need to catch and throw our own exception.